File: docs/WidgetMill/packages/spec/src/nodes/card.ts
import { z } from "zod";
import { BorderRadius } from "../complex/borderRadius";
import { EdgeInsets } from "../complex/edgeInsets";
import { NodeRef } from "../node-ref";
import { BindableColor } from "../scalars";

/** Card — superfície elevada do Material (catálogo Flutter). */
export const CardNode = z.object({
  type: z.literal("card"),
  props: z
    .object({
      color: BindableColor.optional(),
      elevation: z.number().optional(),
      borderRadius: BorderRadius.optional(),
      margin: EdgeInsets.optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/card.dart
import 'package:flutter/material.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Card — superfície elevada do Material.
Widget buildCard(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  final br = parseBorderRadius(p['borderRadius']);
  return Card(
    color: parseColor(p['color']),
    elevation: parseDouble(p['elevation']),
    margin: parseEdgeInsets(p['margin']),
    shape: br != null ? RoundedRectangleBorder(borderRadius: br) : null,
    child: r.maybeRender(context, node.child),
  );
}
